Add new Helm Chart (#16603)
metabase/metabase: docs/operations-guide/running-metabase-on-kubernetes.md

# Running Metabase on Kubernetes

> This guide will help you install Metabase on Kubernetes using [Metabase Helm chart](https://www.metabase.com/helm/index.yaml).

Reviewer: In this specific case, I wanted to link to the repo, so you'll be able to do 
```
helm repo add metabase https://www.metabase.com/helm/
```

| image.tag                          | controller container image tag                              | latest            |
| image.pullPolicy                   | controller container image pull policy                      | IfNotPresent      |
| fullnameOverride                   | String to fully override metabase.fullname template         | null              |
| config.jetty.host                  | Listening on a specific network host                        | 0.0.0.0           |
| config.database.type               | Backend database type                                       | h2                |
| secrets.config.encryptionKey       | Secret key for encrypt sensitive information into database  | null              |
| secrets.database.connString        | Database connection URI (alternative to the below settings) | null              |
| secrets.database.host              | Database host                                               | null              |
| secrets.database.port              | Database port                                               | null              |

> | secrets.database.dbname            | Database name                                               | null              |

Reviewer 2: ```suggestion
| secrets.database.dbName            | Database name                                               | null              |
```

* Kubernetes 1.4+ with Beta APIs enabled

* [Helm](https://github.com/helm/helm) installed

### How to add Metabase official Helm repo

```bash
$ helm repo add metabase https://www.metabase.com/helm
$ helm repo update
```

### How to run this chart
```bash
$ helm install metabase/metabase --generate-name
```
or, in the case you want to set a custom name:
```bash
$ helm install meta-helm-chart metabase/metabase
```

### Introduction

This chart bootstraps a [Metabase](https://github.com/metabase/metabase) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager. It separates concerns about configs in ConfigMaps and Secrets and also sets up liveliness and readiness probes.

> This helm is highly configurable as it can be connected to any application database, tune Jetty or maximum data warehouse connections by passing configuration parameters (be aware that the default deployment will run with the H2 database which is [strongly discouraged](https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html)).

Reviewer 3: ```suggestion
This helm chart is highly configurable. For example, you can:
- Connect to any application database.
- Tune the Jetty server.
- Pass configuration parameters to Metabase to, for example, set the maximum number of data warehouse connections.

Note that the chart's default deployment will run Metabase with the H2 database, which is [strongly discouraged for production environments](https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html).
```
Should 'configuration parameters' be 'environment variables'? And if so, can we link to the page on environment variables?

Also: at a higher level: See my note in the review about default app DB choice.

Reviewer: yeah! those are env variables in the end

> This helm pulls the `latest` docker tag of the `metabase/metabase` image, but it can be easily configured to pull Metabase Enterprise if needed as well as pinning down the version.

Reviewer 3: ```suggestion
This helm chart pulls the `latest` docker tag of the `metabase/metabase` image, but you can configure the chart to pull Metabase Enterprise, or a specific version.
```

### Uninstalling the Chart

> To uninstall/delete the `meta-helm-chart` deployment:

Reviewer 3: This is assuming  `meta-helm-chart` was the custom name the person supplied, yeah?

Reviewer: yup!
